### ISSUE TITLE
Double panic while running `pyoxidizer build`

### ISSUE THREAD

**Issue by uvatbc** [indygreg/PyOxidizer#356]
I have a python project that has a bunch of imports that I'm trying to make a static binary using pyoxidizer.    
I hit a double panic while running `pyoxidizer build`:     
```
Traceback (most recent call last):
  File "/tmp/bytecode-compiler.YgbMHZIUwg8q/bytecodecompiler.py", line 72, in <module>
    code = compile(source, name, "exec", optimize=optimize_level)
  File "pipenv.patched.yaml2.constructor", line 167
    except TypeError, exc:
                    ^
SyntaxError: invalid syntax
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: Empty }', /home/uv/.cargo/registry/src/github.com-1ecc6299db9ec823/python-packaging-0.3.0/src/bytecode.rs:140:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'write failed: Os { code: 32, kind: BrokenPipe, message: "Broken pipe" }', /home/uv/.cargo/registry/src/github.com-1ecc6299db9ec823/python-packaging-0.3.0/src/bytecode.rs:152:36
stack backtrace:                                                                                                                                                                                                                                                                                                                                                    [9/1899]
   0:     0x560e64518580 - std::backtrace_rs::backtrace::libunwind::trace::h04d12fdcddff82aa
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/../../backtrace/src/backtrace/libunwind.rs:100:5
   1:     0x560e64518580 - std::backtrace_rs::backtrace::trace_unsynchronized::h1459b974b6fbe5e1
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x560e64518580 - std::sys_common::backtrace::_print_fmt::h9b8396a669123d95
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x560e64518580 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he009dcaaa75eed60
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x560e6454175c - core::fmt::write::h77b4746b0dea1dd3
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/fmt/mod.rs:1078:17
   5:     0x560e6450f862 - std::io::Write::write_fmt::heb7e50902e98831c
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/io/mod.rs:1518:15
   6:     0x560e6451b135 - std::sys_common::backtrace::_print::h2d880c9e69a21be9
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x560e6451b135 - std::sys_common::backtrace::print::h5f02b1bb49f36879
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x560e6451b135 - std::panicking::default_hook::{{closure}}::h658e288a7a809b29
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:208:50
   9:     0x560e6451add8 - std::panicking::default_hook::hb52d73f0da9a4bb8
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:227:9
  10:     0x560e6451b8d1 - std::panicking::rust_panic_with_hook::hfe7e1c684e3e6462
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:593:17
  11:     0x560e6451b417 - std::panicking::begin_panic_handler::{{closure}}::h42939e004b32765c
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:499:13
  12:     0x560e64518a3c - std::sys_common::backtrace::__rust_end_short_backtrace::h9d2070f7bf9fd56c
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x560e6451b379 - rust_begin_unwind
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:495:5
  14:     0x560e6453fe11 - core::panicking::panic_fmt::ha0bb065d9a260792
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/panicking.rs:92:14
  15:     0x560e6453fc33 - core::option::expect_none_failed::h7e1dd0a94971eb61
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/option.rs:1268:5
  16:     0x560e6416df12 - <python_packaging::bytecode::BytecodeCompiler as core::ops::drop::Drop>::drop::h7cea671f20a8947d
  17:     0x560e63dfbf9d - core::ptr::drop_in_place::h49f357f5cab32d2b
  18:     0x560e63e07758 - <pyoxidizer::py_packaging::standalone_builder::StandalonePythonExecutableBuilder as pyoxidizer::py_packaging::binary::PythonBinaryBuilder>::to_embedded_python_context::h3832e14f10c9d019
  19:     0x560e63dd5ab5 - pyoxidizer::project_building::build_python_executable::hd11a5e3e2275a178
  20:     0x560e63d58c95 - pyoxidizer::starlark::file_resource::file_manifest_add_python_resource::hf75ef84d01a08e78
  21:     0x560e63d5de92 - pyoxidizer::starlark::file_resource::add_python_resource::hf8bbc72293bf9fc3
  22:     0x560e642f7d4b - <T as starlark::values::TypedValueDyn>::call_dyn::ha472292b37f9a8ac
  23:     0x560e64372969 - starlark::values::Value::call::h71c534441d831661
  24:     0x560e64305383 - <starlark::values::function::WrappedMethod as starlark::values::TypedValue>::call::h49d4b715a8051714
  25:     0x560e642f76aa - <T as starlark::values::TypedValueDyn>::call_dyn::h6187f280b52ecda0
  26:     0x560e64372969 - starlark::values::Value::call::h71c534441d831661
  27:     0x560e643253f1 - starlark::eval::eval_call::h9354fd71d0d66d91
  28:     0x560e64326513 - starlark::eval::eval_expr::h03d51a575d93fe70
  29:     0x560e643328f1 - starlark::eval::eval_block::h826a907b8ac217cd
  30:     0x560e642e7476 - <starlark::eval::def::Def as starlark::values::TypedValue>::call::h3556201dd4d831a5
  31:     0x560e642e194a - <T as starlark::values::TypedValueDyn>::call_dyn::h127bb24516ed1b7c
  32:     0x560e64372969 - starlark::values::Value::call::h71c534441d831661
  33:     0x560e642cbca3 - starlark_dialect_build_targets::starlark_resolve_target::hf4322dc0bf1f3de9
  34:     0x560e642d0327 - starlark_dialect_build_targets::resolve_target::h0960b53774758440
  35:     0x560e642f7d4b - <T as starlark::values::TypedValueDyn>::call_dyn::ha472292b37f9a8ac
  36:     0x560e64372969 - starlark::values::Value::call::h71c534441d831661
  37:     0x560e642d10d7 - starlark_dialect_build_targets::resolve_targets::ha433ba406c9abd07
  38:     0x560e642f7d4b - <T as starlark::values::TypedValueDyn>::call_dyn::ha472292b37f9a8ac
  39:     0x560e64372969 - starlark::values::Value::call::h71c534441d831661
  40:     0x560e64323be1 - starlark::eval::eval_call::h1bbdee38b423c6af
  41:     0x560e64329802 - starlark::eval::eval_expr::h5aad299ca6b62d64
  42:     0x560e6432f47b - starlark::eval::eval_stmt::h9aa00908209a8a13
  43:     0x560e64335a9a - starlark::eval::eval_block::hcff257e807925b8d
  44:     0x560e64335d57 - starlark::eval::eval_module::hdf8e4d8f62fb5ead
  45:     0x560e64335faf - starlark::eval::eval_file::h92e897033d0878ac
  46:     0x560e643bfbd3 - starlark::eval::simple::eval_file::he858a81aae4f8847
  47:     0x560e63d3830b - pyoxidizer::starlark::eval::EvaluationContext::evaluate_file::h79b4163d848661a4
  48:     0x560e63d1cc6a - pyoxidizer::projectmgmt::build::h541c9cfda604e9f8
  49:     0x560e63dd25b9 - pyoxidizer::cli::run_cli::h01d780fe9841be90
  50:     0x560e63e077fa - pyoxidizer::main::h6467dc98f8732930
  51:     0x560e63d6d723 - std::sys_common::backtrace::__rust_begin_short_backtrace::h5a03fa531e7eaee4
  52:     0x560e63d6d759 - std::rt::lang_start::{{closure}}::h40a3bc4abadce391
  53:     0x560e6451bde7 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h57e2a071d427b24c
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/ops/function.rs:259:13
  54:     0x560e6451bde7 - std::panicking::try::do_call::h81cbbe0c3b30a28e
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:381:40
  55:     0x560e6451bde7 - std::panicking::try::hbeeb95b4e1f0a876
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:345:19
  56:     0x560e6451bde7 - std::panic::catch_unwind::h59c48ccb40a0bf20
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panic.rs:396:14
  57:     0x560e6451bde7 - std::rt::lang_start_internal::ha53ab63f88fee728
                               at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/rt.rs:51:25
  58:     0x560e63e09192 - main
  59:     0x7fb30b8e90b3 - __libc_start_main
  60:     0x560e63cc712e - _start
  61:                0x0 - <unknown>
thread panicked while panicking. aborting.
Illegal instruction (core dumped)
```

I'm fairly sure there is a bug behind the double exception, but that can come later.     
Let me know if there is more information I can provide.

**Comment by wkschwartz** [comment]
What version of PyOxidizer are you using? Please run `pyoxidizer --version` and post the result.

**Comment by uvatbc** [comment]
On Mac: 
```
14:29:14 uv@Uv-mb-pro:~$ pyoxidizer --version
PyOxidizer 0.10.3
commit: unknown
source: https://github.com/indygreg/PyOxidizer.git
pyembed crate location: version = "0.10.3"
```

On Linux:
```
14:29:47 uv@uv-x1:~$ pyoxidizer --version
PyOxidizer 0.10.3
commit: unknown
source: https://github.com/indygreg/PyOxidizer.git
pyembed crate location: version = "0.10.3"

14:29:52 uv@uv-x1:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
```

**Comment by wkschwartz** [comment]
This report appears to be a duplicate of #147.

**Comment by uvatbc** [comment]
The double panic may be #147, but the stacktrace on the pyoxidizer side is different:     
Frame 16 onwards in this issue (#356) has different signatures as compared to frame 9 onwards on #147

**Comment by wkschwartz** [comment]
Can you try this on `main`? Just in case you haven't installed a development version before, here's how to get commit e86b2f46ed6b449bdb912900b0ac83576ad5ebe9, which is `main` as of the time of this writing:

```console
$ cargo install pyoxidizer --git https://github.com/indygreg/PyOxidizer.git --rev e86b2f46ed6b449bdb912900b0ac83576ad5ebe9
```

Takes about 10-15 minutes to compile, during which time your computer may slow down.

**Comment by indygreg** [comment]
On the `main` branch, bytecode generation failures result in errors like the following:

```
error[PYOXIDIZER_PYTHON_EXECUTABLE]: adding PythonExecutable to FileManifest

Caused by:
    0: building Python executable
    1: building executable with Rust project
    2: obtaining embedded python context
    3: converting lib2to3.tests.data.bom to resource
    4: compiling in-memory bytecode
    5: compiling error: Missing parentheses in call to 'print'. Did you mean print("BOM BOOM!")? (lib2to3.tests.data.bom, line 2)
   --> /home/gps/tmp/myapp/pyoxidizer.bzl:270:5
    |
270 |     files.add_python_resource(".", exe)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PythonExecutable.to_file_manifest()
```

I believe the panic was fixed by ec8aa6986.

Since the remaining part of this issue is #147, I'm going to close this issue.
